pkg/util/util.go

```diff
@@ -358,3 +368,54 @@ func Md5sum(filePath string) (string, error) {
 	hashInBytes := hash.Sum(nil)[:16]
 	return hex.EncodeToString(hashInBytes), nil
 }
+
+// Three functions for zeroing a range in the destination file:
+
+// PunchHole attempts to zero a range in a file with fallocate, for block devices and pre-allocated files.
+func PunchHole(outFile *os.File, start, length int64) error {
```

Comment:
Do we need some unit tests for these?

Reply:
Yes, I should have done this in the first place. I also renamed some of them to be a little clearer about what they do.